BUG-49659 CLOB 데이터 타입 조회가 필요합니다. (#200)
ALTIBASE/Documents: Manuals/Tools/Altibase_trunk/eng/Altibase 3rd Party Connector Guide.md

![](media/3rdPartyConnector/9282c65c5cdf53f32d920ae18dfbe692.jpg)

### FAQ

#### After installing the latest JDK version, when running the SQuirreL SQL client, it fails with the message "Your Java Virtual Machine must be at least 1.6 to run SQuirrel 3.x and above". 

This error occurs because the SQuirreL SQL client does not recognize the latest Java version.

Find the JavaVersionChecker line in the squirrel-sql.bat or squirrel-sql.sh file and add the version to use. For example, when using OpenJDK 18 version, add '18' at the end of the JavaVersionChecker line and save it, then run the SQuirreL SQL client.

`$JAVACMD -cp "$UNIX_STYLE_HOME/lib/versioncheck.jar" JavaVersionChecker 1.6 1.7 1.8 18`

Please refer to https://sourceforge.net/p/squirrel-sql/bugs/1347/

> #### How to lookup LOB data.

Reviewer: ```suggestion
#### How to retrieve LOB data.
```

Author: 반영했습니다.

> In the SQuirreL SQL client, LOB data can be viewed in the SQL tab or the Objects tab.

Reviewer: 이 문자에서 view 도 다른 단어를 사용하는 것이 어떨까 싶습니다. 

영문을 보다보니, 영문이 아래 문장으로 작성되면 좋을 것 같습니다. 

SQuirreL SQL 클라이언트에서 LOB 데이터 조회는 SQL 탭 또는 Objects 탭을 이용해야 한다.
또는
SQuirreL SQL 클라이언트에서 LOB 데이터 조회는 SQL 탭 또는 Objects 탭에서 할 수 있다.

> In the SQuirreL SQL client, LOB data can be viewed in the SQL tab or the Objects tab.

Reviewer: ```suggestion
In SQuirrel SQL client, LOB data can be retrieved in SQL tab or Objects tab.
```

Author: the 필요할 것 같습니다.

Author: the는 필요할 것 같습니다.

Reviewer: > the는 필요할 것 같습니다.

넵

| SQL                                               | Objects                                               |
| ------------------------------------------------- | ----------------------------------------------------- |
| ![](media/3rdPartyConnector/squirrel_lob_sql.jpg) | ![](media/3rdPartyConnector/squirrel_lob_objtree.jpg) |

> ##### Settings for LOB data inquiry in SQuirreL SQL client

Reviewer: 구글링해보면 inquiry는 잘 안 사용하는 것 같습니다. retrieve 이나 fetch 가 어떨까요?

> ##### Settings for LOB data inquiry in SQuirreL SQL client

Reviewer: ```suggestion
##### Settings for LOB data retrieval in SQuirreL SQL client
```

> This setting is a commonly required setting when searching for LOB data in the SQL tab or in the Objects tab.

Reviewer: 한글 문장을 아래와 같이 변경해서 수정해보았습니다. 

이 설정은 SQL 탭이나 개체 탭에서 LOB 데이터를 검색할 때 필요하다.

```suggestion
This setting is required when retrieving LOB data from the SQL tab or the Object tab. 
```

> This setting is a commonly required setting when searching for LOB data in the SQL tab or in the Objects tab.

Reviewer: ```suggestion
This setting is required when retrieving LOB data from the SQL tab or the Object tab. 
```

Author: Oracle쪽 참조했는데, from 보다는 in 이 더 적합하지 않을까 합니다. 
https://docs.oracle.com/cd/E25259_01/appdev.31/e24285/intro.htm#RPTUG10720

아니면 아래와 같이 바꿔도 괜찮을 것 같구요. 버그의 목적이 SQuirrel SQL 설정을 바꾸어서 Altibase LOB  데이터를 볼 수 있게 하는 겁니다.
The following setting is required to display LOB data in the SQL tab or the Object tab.

Author: 아..그리고 탭 이름 자체가 "Objects" 입니다.

Reviewer: > The following setting is required to display LOB data in the SQL tab or the Object tab.

네. 알겠습니다.

1. Change the Data Type Controls setting.<br>

>    In 'File -> Global Preferences -> Data Type Controls tab', check the "Read contents where table is first loaded" option of BLOB or CLOB 

Reviewer 2: where table is first loaded => when table is first loaded
이것 포함 5군데 모두 수정되어야 할 것 같습니다.

Author: 반영하도록 하겠습니다.

![](media/3rdPartyConnector/squirrel_lob_view.jpg)

2. Change the autocommit mode to false in the session properties.<br>
   In Altibase HDB, transactions must be managed explicitly for LOB data. To apply this, uncheck the "Auto Commit SQL" checkbox in File -> New Session Properties -> SQL tab.

![](media/3rdPartyConnector/squirrel_lob_autocommit.jpg)

> ##### Additional required settings when looking up LOB data in the Objects tab

Reviewer: 여기 look up 도 다른 단어로...

> ##### Additional required settings when looking up LOB data in the Objects tab

Reviewer: ```suggestion
##### Additional required settings when retrieving LOB data in the Objects tab
```

![](media/3rdPartyConnector/9282c65c5cdf53f32d920ae18dfbe692.jpg)

### FAQ

#### After installing the latest JDK, when running the SquirreL SQL client, it fails with the message "Your Java Virtual Machine must be at least 1.6 to run SQuirrel 3.x and above". 

This error occurs because the SquirreL SQL client does not recognize the latest Java version.

> Find the JavaVersionChecker line in the squirrel-sql.bat or squirrel-sql.sh file and add the version to use. For example, when using OpenJDK 18 version, 

Reviewer 3: 문장 마지막에 'find the line below'처럼 예시를 더 직접적으로 가르키는 문장이 있었으면 좋겠습니다.

Reviewer 3: 에러 메시지에서 SQuirrel -> Squirrel 수정해야 할 것 같습니다.

Author: 1. 문장 마지막에 'find the line below'처럼 예시를 더 직접적으로 가르키는 문장이 있었으면 좋겠습니다.
-> 좀더 구체적으로 부탁드려도 될까요?

2. 제품명이 정확히는 SQuirrel이 맞습니다.

Reviewer: > 1. 문장 마지막에 'find the line below'처럼 예시를 더 직접적으로 가르키는 문장이 있었으면 좋겠습니다.
>    -> 좀더 구체적으로 부탁드려도 될까요?
> 2. 제품명이 정확히는 SQuirrel이 맞습니다.

이번에 전체적으로 SQuirreL SQL 로 변경 부탁드려도 될까요?

Author: 일괄 변경했습니다.

Reviewer 3: > 1. 문장 마지막에 'find the line below'처럼 예시를 더 직접적으로 가르키는 문장이 있었으면 좋겠습니다.
>    -> 좀더 구체적으로 부탁드려도 될까요?
> 2. 제품명이 정확히는 SQuirrel이 맞습니다.

해당 문장 맨 마지막 부분에 'find the line below'라는 구절을 추가하는 게 좋겠다는 의견이었습니다.
바로 아래에서 수정 전 문장을 보여주고, 그 다음에 18을 추가한 문장을 보여주어서입니다.

Author: 희정님 의견 반영해, 라인을 합쳐 하나의 문장으로 연결했습니다. 변경으로 인해 'find the line below' 반영은 불필요할 것 같습니다.

`$JAVACMD -cp "$UNIX_STYLE_HOME/lib/versioncheck.jar" JavaVersionChecker 1.6 1.7 1.8`

> add 18 at the end of the JavaVersionChecker line  and save it, and then run the SquirreL SQL client.

Reviewer 3: add '18' at the end of the JavaVersionChecker line just as below and save it, then run the SquirreL SQL client.

Author: 적용했습니다.

`$JAVACMD -cp "$UNIX_STYLE_HOME/lib/versioncheck.jar" JavaVersionChecker 1.6 1.7 1.8 18`

> 참조: https://sourceforge.net/p/squirrel-sql/bugs/1347/

Reviewer 3: Please refer to https://sourceforge.net/p/squirrel-sql/bugs/1347/

Author: 적용했습니다.

In the SquirreL SQL client, LOB data can be viewed in the SQL tab or the Objects tab.

| SQL                                               | Objects                                               |
| ------------------------------------------------- | ----------------------------------------------------- |
| ![](media/3rdPartyConnector/squirrel_lob_sql.jpg) | ![](media/3rdPartyConnector/squirrel_lob_objtree.jpg) |

##### Settings for LOB data lookup in SQL tab.

1. Change the Data Type Controls setting.
   In 'File -> Global Preferences -> Data Type Controls tab', check the "Read contents where table is first loaded" option of BLOB or CLOB 

![](media/3rdPartyConnector/squirrel_lob_view.jpg)

2. Change the autocommit mode to false in the session properties.
   In Altibase HDB, transactions must be managed explicitly for LOB data. To apply this, uncheck the "Auto Commit SQL" checkbox in File -> New Session Properties -> SQL tab.

![](media/3rdPartyConnector/squirrel_lob_autocommit.jpg)

##### Additional required settings when looking up LOB data in the Objects tab

1. Use the Altibase HDB JDBC Driver corresponding to the version below.
   Altibase 7.1.0.7.2 or higher

>    Altibase 7.2.0.0.2 or higher

Reviewer 3: Altibase 7.1.0.7.2 or higher, Altibase 7.2.0.0.2 or higher 둘 중에 하나만 작성해도 되지 않는지 여쭤보고 싶습니다.
또한 줄바꿈하여 불렛 표시를 하려고 하신 건가요? 웹에서는 같은 줄에 표시됩니다.

Author: 아닙니다. 7.1 버전과 7.2 버전 구분해야 합니다.
줄바꿈은 불렛 추가했습니다.
